azurecluster_test.go

```diff
@@ -113,18 +113,29 @@ func Test_AzureClusterCR(t *testing.T) {
 		t.Fatalf("error finding MachinePools for cluster %q: %s", clusterID, microerror.JSON(err))
 	}
 
-	sort.Slice(machinePools, func(i int, j int) bool {
-		return machinePools[i].Name < machinePools[j].Name
+	var readyMachinePools []capiexp.MachinePool
+	for _, machinePool := range machinePools {
+		if capiconditions.IsTrue(&machinePool, capi.ReadyCondition) {
+			readyMachinePools = append(readyMachinePools, machinePool)
+		}
+	}
+
+	if len(readyMachinePools) < 1 {
+		t.Fatal("there are no 'Ready' node pools to test")
```

Comment:
we fail if there are no ready node pools because in that case we wouldn't be testing anything